src/pool.c

```diff
@@ -799,6 +799,29 @@ worker_list(char *list_start, char *list_end, const char *address)
     pthread_rwlock_unlock(&rwlock_acc);
 }
 
+void
+worker_hr(double *avg, const char *address, const char *rig_id)
+{
+    account_t *account = NULL;
+    if (!*rig_id)
```

Comment:
If you're going to check `rig_id`, you may as well check `address` too, and properly:

    if (!address || !rig_id || !*address || !*rig_id)